### PR TITLE
fix: improve IP extraction config editor

### PR DESCRIPTION
**Pull request by ding113**
## Summary
Improves the IP extraction config editor in system settings to always display the effective configuration (either custom or default) and enhances the UX with a tooltip-based help system.

## Problem
The IP header extraction configuration editor had several usability issues:
- When no custom value was stored, the textarea appeared empty, making it unclear what the actual default configuration was
- The long help text for the extraction chain was always visible, cluttering the UI
- The "Reset to default" behavior wasn't clear about whether it would immediately save or just populate the field

## Solution
- Always show the effective IP header extraction JSON (custom if set, otherwise the built-in default) in the textarea
- Move the lengthy extraction-chain guidance into an accessible tooltip (CircleHelp icon) next to the label
- Add localized tooltip label strings for all 5 supported languages
- Change "Reset to default" to only insert the default JSON into the textarea without saving, giving users a chance to review before explicitly saving
- Add the default configuration as a textarea placeholder for additional clarity

### Core Changes (src/app/[locale]/settings/config/_components/system-settings-form.tsx)
- Added `formatIpExtractionConfig()` helper to consistently format JSON with indentation
- Added `DEFAULT_IP_EXTRACTION_CONFIG_TEXT` constant for reuse
- Changed initial state to always show formatted default config when no custom value exists
- Added Tooltip component wrapping a help button with CircleHelp icon
- Changed Reset button behavior to populate textarea with default JSON instead of clearing it
- Added `placeholder` attribute showing the default config

### i18n Updates (messages/*/settings/config.json)
- Added `extractionConfigHelpLabel` translation key for tooltip button accessibility in all 5 languages (en, ja, ru, zh-CN, zh-TW)

### Tests (tests/unit/settings/system-settings-form-ip-extraction.test.tsx)
- **New test file** (241 lines) covering:
  - Default config display when no custom value is stored
  - Reset to default button behavior (only populates, doesn't auto-save)
  - Form submission with default JSON when no changes made
  - Form submission with `null` when user explicitly clears the textarea

### Automated Tests
- [x] Unit tests added (`system-settings-form-ip-extraction.test.tsx`)
- [x] Existing IP extraction tests verified (`extract-client-ip.test.ts`)

### Manual Testing Performed
1. Open System Settings → Config page
2. Verify IP extraction config textarea shows formatted default JSON
3. Hover over help icon to verify tooltip displays correctly
4. Click "Reset to default" and verify textarea populates with default JSON
5. Modify the JSON and save, verify custom config persists
6. Clear textarea and save, verify it saves as null (reverts to system default)

## Checklist
- [x] Code follows project conventions (Biome formatting)
- [x] Self-review completed
- [x] Tests pass locally (Vitest)
- [x] TypeScript checks pass (tsgo)
- [x] Build passes (Next.js production build)
- [x] i18n strings added for all 5 languages
- [x] No breaking changes (UI enhancement only)

---
*Original description preserved - enhanced for reviewer clarity*

<details open><summary><h3>Greptile Summary</h3></summary>

This PR improves the IP extraction config editor by pre-filling the textarea with the effective configuration (custom or built-in default), replacing the always-visible help text with a tooltip (`CircleHelp` icon), changing "Reset to default" to populate without auto-saving, and adding the `extractionConfigHelpLabel` i18n key to all five locales. The UX improvements are clear and well-tested, though two concerns raised in prior review rounds remain open: (1) pre-filling the textarea when `ipExtractionConfig` is `null` means the very first save on an unconfigured instance will write the current hardcoded default to the DB, blocking future upgrades from changing that default; (2) the shape validator does not verify that each header entry has the required `name: string` field.
</details>

<details open><summary><h3>Confidence Score: 3/5</h3></summary>

Not safe to merge until the implicit-null-to-explicit-default write issue is resolved.

The two P1 findings from prior review threads — implicit promotion of null to the hardcoded default on any save, and incomplete per-item shape validation — are still present and unaddressed. The new test at line 213 actually asserts the implicit-write behavior as expected, confirming it has not been fixed.

src/app/[locale]/settings/config/_components/system-settings-form.tsx lines 179-181 (initial state) and 215-223 (shape validator) need attention before merging.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/app/[locale]/settings/config/_components/system-settings-form.tsx | Core form component: adds helper/constant, pre-fills textarea with default when config is null, tooltip help button, and updated reset logic — but pre-filling means any save while config is null writes the hardcoded default to the DB (prior P1 thread), and the shape validator doesn't check that each header item has the required `name` field (prior P1 thread). |
| tests/unit/settings/system-settings-form-ip-extraction.test.tsx | New 241-line unit test covering default display, reset-to-default, and save behavior; test 3 explicitly asserts that saving when config is null submits DEFAULT_IP_EXTRACTION_CONFIG (confirming the implicit-write behavior). Error messages and describe/test labels are in Chinese only. |
| messages/en/settings/config.json | Adds `extractionConfigHelpLabel` key to the `ipLogging` namespace; all 5 locales updated consistently. |

</details>

</details>

<details open><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant DB as Database
    participant Form as SystemSettingsForm
    participant UI as Textarea

    Note over Form: Mount / initialSettings loaded
    DB-->>Form: ipExtractionConfig = null
    Form->>UI: value = DEFAULT_IP_EXTRACTION_CONFIG_TEXT

    Note over UI: User makes no changes and saves
    UI->>Form: ipExtractionConfigText = default JSON
    Form->>DB: saveSystemSettings({ ipExtractionConfig: DEFAULT_IP_EXTRACTION_CONFIG })
    Note over DB: null → hardcoded default written permanently

    Note over UI: User explicitly clears textarea and saves
    UI->>Form: ipExtractionConfigText = empty
    Form->>DB: saveSystemSettings({ ipExtractionConfig: null })
    DB-->>Form: result.data.ipExtractionConfig = null
    Form->>UI: value = DEFAULT_IP_EXTRACTION_CONFIG_TEXT (re-filled)
    Note over UI: UI shows default even though null was saved
```
</details>

<details><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `src/app/[locale]/settings/config/_components/system-settings-form.tsx`, line 215-223 ([link](https://github.com/ding113/claude-code-hub/blob/8d588b5b2e76fac34fdd46e878c635475351feb4/src/app/[locale]/settings/config/_components/system-settings-form.tsx#L215-L223)) 

   <a href="#">P2</a> **Incomplete header-item validation**

   The shape check only confirms `headers` is an array — it does not verify that each element contains the required `name: string` field from `IpHeaderRule`. Input like `{"headers":[{"pick":"leftmost"}]}` (missing `name`) would pass this guard, be persisted to the DB, and silently break IP extraction at runtime.

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: src/app/[locale]/settings/config/_components/system-settings-form.tsx
   Line: 215-223

   Comment:
   **Incomplete header-item validation**

   The shape check only confirms `headers` is an array — it does not verify that each element contains the required `name: string` field from `IpHeaderRule`. Input like `{"headers":[{"pick":"leftmost"}]}` (missing `name`) would pass this guard, be persisted to the DB, and silently break IP extraction at runtime.

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: tests/unit/settings/system-settings-form-ip-extraction.test.tsx
Line: 139-167

Comment:
**Test error messages and names in Chinese only**

All `describe`/`test` labels and every internal `throw new Error(...)` in this file are written in Chinese. For an open-source project with international contributors, Chinese-only test output makes failures hard to triage without translation. Consider using English (matching the rest of the codebase's test files) or at least providing bilingual messages.

```suggestion
function getIpExtractionTextarea() {
  const textarea = document.getElementById("ip-extraction-config") as HTMLTextAreaElement | null;
  if (!textarea) throw new Error("IP extraction config textarea not found");
  return textarea;
}
```

The same applies to the helper that looks up a button by text (`"Button not found: ${text}"`) and the one that locates the form element (`"System settings form not found"`).

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (2): Last reviewed commit: ["test: clarify IP extraction form fixture..."](https://github.com/ding113/claude-code-hub/commit/2d9de44ace9cc790e5e650a493beecf83ac449b4) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=28898036)</sub>

### CODE REVIEW

File: src/app/[locale]/settings/config/_components/system-settings-form.tsx [hunk 1]
```diff
@@ -169,9 +177,7 @@ export function SystemSettingsForm({ initialSettings }: SystemSettingsFormProps)
     initialSettings.ipGeoLookupEnabled ?? true
   );
   const [ipExtractionConfigText, setIpExtractionConfigText] = useState<string>(
-    initialSettings.ipExtractionConfig
-      ? JSON.stringify(initialSettings.ipExtractionConfig, null, 2)
-      : ""
+    formatIpExtractionConfig(initialSettings.ipExtractionConfig ?? DEFAULT_IP_EXTRACTION_CONFIG)
```
[the remark this answers is not in the document]
Comment (ding113) [reply src/app/[locale]/settings/config/_components/system-settings-form.tsx:180]: 已复核，这里没有改回空值，因为这是本 PR 已确认的产品语义：输入框默认直接展示当前生效 JSON；无自定义时展示默认链；点击“恢复默认”插入默认 JSON；保持该默认 JSON 并保存时，数据库显式保存默认对象。只有用户手动清空输入框时才提交 null。这个显式持久化行为是有意设计，不是无意回归。

File: src/app/[locale]/settings/config/_components/system-settings-form.tsx [hunk 2]
```diff
@@ -283,9 +289,7 @@ export function SystemSettingsForm({ initialSettings }: SystemSettingsFormProps)
         );
         setIpGeoLookupEnabled(result.data.ipGeoLookupEnabled ?? true);
         setIpExtractionConfigText(
-          result.data.ipExtractionConfig
-            ? JSON.stringify(result.data.ipExtractionConfig, null, 2)
-            : ""
+          formatIpExtractionConfig(result.data.ipExtractionConfig ?? DEFAULT_IP_EXTRACTION_CONFIG)
```
[the remark this answers is not in the document]
Comment (ding113) [reply src/app/[locale]/settings/config/_components/system-settings-form.tsx:292]: 这里同上：保存成功后的 UI 需要继续显示当前正在生效的 JSON。若服务端返回 null，按本次确认的产品语义会回显为格式化后的默认链，而不是空串；手动清空输入框再保存时仍会提交 null。

File: src/app/[locale]/settings/config/_components/system-settings-form.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ding113) [reply src/app/[locale]/settings/config/_components/system-settings-form.tsx:180]: 已核对，这里确实会把“展示中的默认 JSON”持久化为显式对象；这是本 PR 在实现前已经锁定的产品决定，而不是遗漏的兼容性问题。相应地，测试里也分别锁定了“默认 JSON 保存为对象”和“手动清空保存为 null”两条语义。

File: src/app/[locale]/settings/config/_components/system-settings-form.tsx [hunk 3]
```diff
@@ -1013,25 +1017,41 @@ export function SystemSettingsForm({ initialSettings }: SystemSettingsFormProps)
...
               className={`${inputClassName} font-mono text-xs`}
             />
-            <p className="text-xs text-muted-foreground">{tIpLogging("extractionConfigHint")}</p>
             <div>
               <Button
                 type="button"
                 variant="outline"
                 size="sm"
-                onClick={() => setIpExtractionConfigText("")}
+                onClick={() => setIpExtractionConfigText(DEFAULT_IP_EXTRACTION_CONFIG_TEXT)}
```
[the remark this answers is not in the document]
Comment (ding113) [reply src/app/[locale]/settings/config/_components/system-settings-form.tsx:1054]: 这里没有改成清空字段，是因为需求已经明确指定“恢复默认”按钮应当把默认值插入输入框，而不是立刻改库或回退为空。当前行为与该产品要求一致；若用户希望恢复到 null 语义，仍可手动清空后保存。

File: src/app/[locale]/settings/config/_components/system-settings-form.tsx [hunk 4]
```diff
@@ -169,9 +177,7 @@ export function SystemSettingsForm({ initialSettings }: SystemSettingsFormProps)
     initialSettings.ipGeoLookupEnabled ?? true
   );
   const [ipExtractionConfigText, setIpExtractionConfigText] = useState<string>(
-    initialSettings.ipExtractionConfig
-      ? JSON.stringify(initialSettings.ipExtractionConfig, null, 2)
-      : ""
+    formatIpExtractionConfig(initialSettings.ipExtractionConfig ?? DEFAULT_IP_EXTRACTION_CONFIG)
   );
```
[the remark this answers is not in the document]
Comment (ding113) [reply src/app/[locale]/settings/config/_components/system-settings-form.tsx:181]: 这个点我核对过，行为是有意的：本次需求要求“默认直接显示当前正在生效的 JSON（无自定义则显示默认值）”，并且在规划确认时已明确选择“直接保存默认 JSON 对象”。因此这里没有按“保留 null 语义”去实现。

File: tests/unit/settings/system-settings-form-ip-extraction.test.tsx [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    await submitForm();
+
+    expect(systemConfigActionMocks.saveSystemSettings).toHaveBeenCalledWith(
+      expect.objectContaining({
+        ipExtractionConfig: DEFAULT_IP_EXTRACTION_CONFIG,
+      })
+    );
+
+    unmount();
+  });
```
[the remark this answers is not in the document]
Comment (ding113) [reply tests/unit/settings/system-settings-form-ip-extraction.test.tsx:225]: 这个 nitpick 成立，已在 commit 2d9de44a 调整：把 baseSettings 放到 buildSettings 前面，避免在声明上方引用下方常量类型。业务断言保持不变，因为“未修改即保存默认对象”是本 PR 的既定语义。
